### PR TITLE
Remove cython as run dependency

### PR DESCRIPTION
**Pull request by DManowitz**
Remove cython dependencies from host & run sections.  Also, remove `build_platform != target_platform` checks from python, cython, and numpy dependencies in build section.

Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

Addresses #67

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,32 +8,31 @@ source:
...
   sha256: d398acb69e0c4ebdf539c2ba5839c523360ec814f300faa7f2f87de8f257af58
 build:
-  number: 0
+  number: 1
   script: {{ PYTHON }} -m pip install . --no-deps -vv
 
 requirements:
   build:
-    - python                                 # [build_platform != target_platform]
+    - python
```
Comment (jakirkham) [inline recipe/meta.yaml:16]:
```suggestion
    - python                                 # [build_platform != target_platform]
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -8,32 +8,31 @@ source:
...
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - cython >=0.27,<3                       # [build_platform != target_platform]
-    - numpy                                  # [build_platform != target_platform]
+    - cython >=0.27,<3
```
Comment (jakirkham) [inline recipe/meta.yaml:18]:
```suggestion
    - cython >=0.27,<3                       # [build_platform != target_platform]
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -8,32 +8,31 @@ source:
...
   host:
     - python
     - pip
+    - cython >=0.27,<3    
     - setuptools
     - scikit-learn >=0.16
-    - cython >=0.27,<3
     - joblib
-    - numpy
+    - {{ pin_compatible('numpy') }}
```
Comment (jakirkham) [inline recipe/meta.yaml:29]:
```suggestion
    - numpy
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -8,32 +8,31 @@ source:
...
 
 requirements:
   build:
-    - python                                 # [build_platform != target_platform]
+    - python
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - cython >=0.27,<3                       # [build_platform != target_platform]
-    - numpy                                  # [build_platform != target_platform]
+    - cython >=0.27,<3
+    - {{ pin_compatible('numpy') }}
```
Comment (jakirkham) [inline recipe/meta.yaml:19]:
```suggestion
    - numpy                                  # [build_platform != target_platform]
```

**Review by jakirkham — COMMENTED** [review]
Thanks David! 🙏

> Also, remove build_platform != target_platform checks from python, cython, and numpy dependencies in build section.

These are there because those packages are used for cross-compilation. Would recommend keeping those checks as they were

**Comment by DManowitz** [comment]
> Thanks David! 🙏
> 
> > Also, remove build_platform != target_platform checks from python, cython, and numpy dependencies in build section.
> 
> These are there because those packages are used for cross-compilation. Would recommend keeping those checks as they were

I guess I'm a little confused as to why those packages wouldn't be needed for compilation on ALL platforms, not JUST for cross-compilation.

It appears that with my second commit, the only failing build is `linux linux_64_numpy1.26python3.12.__cpython`, which I will look into.

**Comment by DManowitz** [comment]
> It appears that with my second commit, the only failing build is `linux linux_64_numpy1.26python3.12.__cpython`, which I will look into.

This is a timeout issue in the build.  Is there a way to re-trigger this build?

**Comment by DManowitz** [comment]
> The cross-compilation case differs as the binaries in `host` are not runnable on the `build` platform where the build occurs. So we actually need an additional set of packages that run on the build machine to drive the build process

Ok.  That answers the question in terms of python and numpy.  However, part of my initial question is whether it makes sense to keep Cython as a `host` dependency, since once the package is built, it would seem to be unnecessary.

**Comment by jakirkham** [comment]
The `build` & `host` dependencies are only used at build time. It is ok if they only appear there (as `cython` typically does)

**Comment by jakirkham** [comment]
LGTM

Could someone from @conda-forge/hdbscan also review? 🙂

**Comment by DManowitz** [comment]
@johnlees @lmcinnes Can either of you please review this?  Thanks.

**Comment by DManowitz** [comment]
Closing because already late by one version, so redoing with latest hdbscan version.

**Pull request closed without merging**